api/src/main/java/org/eclipse/microprofile/concurrent/ManagedExecutorConfig.java

```diff
@@ -19,27 +19,39 @@
 package org.eclipse.microprofile.concurrent;
 
 import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
 /**
- * <p>Configures an injected <code>ManagedExecutor</code> instance.</p>
+ * <p>Provides configuration for an injected {@link ManagedExecutor} instance
+ * which is {@link javax.enterprise.context.Dependent dependent} scoped.
+ * When an application has multiple injection points for {@link ManagedExecutor}
+ * with matching configuration, the container must inject the same instance.
```

Comment:
If the application is to rely on the concurrency constraints provided by the managed executor configuration, it needs a guarantee that those constraints will be enforced and will not vary by implementation.  This targets the most common pattern, but I do agree that a valid pattern also exists where an application might want duplicate executors with identical configuration.  For this less common case, they would currently only be able to create one of those via CDI and would need to rely on the builder to create the other (with a producer, they could make it available in CDI as well).  Another way to address this could be to add a 'name' attribute to the configuration that would allow for arbitrary distinction between instances. 
`@Inject @ManagedExecutorConfig(name="high-priority-work", maxAsync=5)`
`@Inject @ManagedExecutorConfig(name="low-priority-work", maxAsync=5)`
 Do you have any insights on how this is typically addressed in CDI?

Reply:
The name would work, inside the producer methods providing this bean, you could determine whether you create a new one or return an existing instance. Not sure if it is worth the hassle though, we probably don't have use case for this yet.
So we could just keep a GH issue on the side track and drop the idea for now.